src/pvrclient-nextpvr.cpp

```diff
@@ -431,8 +431,10 @@ PVR_ERROR cPVRClientNextPVR::GetChannelStreamProperties(const kodi::addon::PVRCh
       kodi::Log(ADDON_LOG_ERROR, "Transcoding Error");
       return PVR_ERROR_FAILED;
     }
+    properties.emplace_back(PVR_STREAM_PROPERTY_INPUTSTREAM, PVR_STREAM_PROPERTY_VALUE_INPUTSTREAMFFMPEG);
```

Comment:
This needs to be: 
`properties.emplace_back(PVR_STREAM_PROPERTY_INPUTSTREAM, "inputstream.ffmpegdirect");`

You have set it to use kodi's internal ffmpeg inputstream and not the ffmpegdirect addon.

Reply:
You may also need to set these two properties:

`
properties.emplace_back("inputstream.ffmpegdirect.is_realtime_stream", "true");
properties.emplace_back("inputstream.ffmpegdirect.manifest_type", "hls");
`

Although it may work without them.

Reply:
Thanks a lot I didn't know there where two but now it makes sense.

Unfortunately that crashes almost immediately after playback in void CDVDDemuxUtils::FreeDemuxPacket(DemuxPacket* pPacket)  The Addon does create a 0 byte seg/idx file.

Here is the debug log https://paste.kodi.tv/ecisuyafij.kodi

Reply:
Can you provide a backtrace with a debugger?

Reply:
Can you provide a backtrace with a debugger?

Reply:
Does this help?
>
```
 	kodi.exe!av_packet_free_side_data()	C
>	kodi.exe!CDVDDemuxUtils::FreeDemuxPacket(DemuxPacket * pPacket) Line 32	C++
	kodi.exe!CInputStreamAddon::cb_free_demux_packet(void * kodiInstance, DemuxPacket * packet) Line 663	C++
```

Reply:
Anyway you can get me access to a test stream?

Reply:
Would the actual set of files help  https://1drv.ms/u/s!Ar-w3NVk_f-MlUxPnO67Uv7BVd4e?e=tOWCp0   A transcoded file is created on the fly by the nextpvr backend.

The logs for the thread are here https://paste.kodi.tv/mojusoyazi.kodi

Reply:
Actually cool it is working now with the nightly,  my self debug compile of kodi might be the issue, I will check the dependencies.

It works great but here are two issues I am having

1) - perhaps I am missing a setting I see lots of spam with

2020-06-29 10:11:47.264 T:54400   DEBUG <CAddonSettings[inputstream.ffmpegdirect]>: loading setting definitions

2) - the arrow  button doesn't take me to the beginning of the buffer, it takes me to random place.  I can skip on the timeline and with the mouse and cursor.   https://youtu.be/6CZ0zYB9WfU

Reply:
1. That's not the addon. It's the debug code in the settings manager.

2. Yes, that's correct. The left arrow will try to take you to the start of the programme which is not possible. It works as expected once the next programme starts.

Reply:
> 
>     2. Yes, that's correct. The left arrow will try to take you to the start of the programme which is not possible. It works as expected once the next programme starts. I tried to add a correction there for the seek but no matter what I tried I couldn't get it to work.

Do you set the SetStartTime() to clock time on the first play?

I think this is a really nice option for users of NextPVR that need transcoding remote and have the system space.  It is the next  best thing to skipping on directly via the m3u8 file.  Thanks for this feature and the help getting it going.

Reply:
Yes, it’s a different structure in an inputstream but similar to `GetStreamTimes` in PVR.

https://github.com/xbmc/inputstream.ffmpegdirect/blob/2d485c3b3a3db982be81466bc34df07508cb978d/src/stream/TimeshiftStream.cpp#L125

Reply:
The windows crashing problem no longer is an issue after compiling a debug version of the inputsream.ffmpegdirect so this  is resolved.

Reply:
All issues resolved with https://github.com/xbmc/inputstream.ffmpegdirect/pull/59